### ISSUE TITLE
[Suggestion]Share texture between models

### ISSUE THREAD

**Issue by Tunied** [simlu/voxelshop#223]
Hi. 
Voxelshop is really really great tool.   i already use it in my project. batter than MagicVoxel as it has layer support!

during my use, i foud out for Voxelshop, there is no way to share textrue(color paletter). during models.

ex: i have 10 models and one color palette with 100 color in it.  

i can't find an way to export those 10 models with the same color paletter.  it seem only export the color they are using .  

so is this possible to add an option that can export the texture use the whole color palette?

thanks.

**Comment by simlu** [comment]
@Tunied Are you talking about exporting to MagicaVoxel and the color palette that is exported with that?

**Comment by Tunied** [comment]
Hi 

for now , my work flow is model the character in VoxelShop . each part is in separate layer 
eg. Hand,Leg,Head etc.

and animation it with rig in 3DMax, and export the rig animtion to Unity. 

also , i use the export dae file from VoxelShop.

================

my game logic allow player to change character part. 

so for an human  will have one animation file(riged in 3DMax and export to unity)

and some sperate modles to replace . like  

Head x 10 
Leg x 5
Hand x 7.

all those modles i use the same color paletter (there will be 100 color in it.),

normally each modle will have there own texture ,  then texture only include the color they are using in the modle. 

but since all those molde are use the same color paletter.  so if VoxelShop can provied an export operation like use the entire color paletter as the texture. then i can relink all modle texture in Unity. 

let them use the same one . so that can drop some memory use , also it would be easy to change color style later.


thanks.

**Comment by simlu** [comment]
Can't you just put them into one file and then export?

**Comment by Tunied** [comment]
not really...  like sword, guns...   if put everything in one file . it will be chaos...

**Comment by simlu** [comment]
Do you mean texture when you write color palette btw?

**Comment by simlu** [comment]
Im assuming you are exporting as collada with texture but group by color? In that case it might be possible to do what you're asking as an option.

Why don't you use vertex coloring?

**Comment by simlu** [comment]
Well... not really. The color palette has really nothing to do with the rest of the program. It's just a way to store a set of colors. So there can be other colors in the model that are not in the color palette itself.

**Comment by Tunied** [comment]
sorry not reply soon, busy on something else :(

> Sounds like the problem is, the exported 'layers' are exported with a subset of the global color palette. is that correct?

yes, that it what i mean. 

and i also fond some confuse on export multiple layer modle , the pivot point seem not what i expect it should be.

ex: i have modle with layer head,foot,body, when i export it to 3dmx , all three part pivot point is at floor
point (0,0,0).  that's will be a little problem when i try to animated it in Unity or 3DMax.

anyway i found i way to work around it .

first i modle it in Voxelshop and export with Qubicle format , and use [this tool](https://chrmoritz.github.io/Troxel/)  to convert it MagicVoxel format.

and colour it also export separate part with Obj format in that tool.

that's will achieve my goal :)

**Comment by kevzettler** [comment]
>anyway i found i way to work around it .

>first i modle it in Voxelshop and export with Qubicle format , and use this tool to convert it MagicVoxel format.

>and colour it also export separate part with Obj format in that tool.

So sounds like you want all the colors you used in VoxelShop, Which get exported individually to Quibicle, to eventually end up in your MagicaVoxel Pallette?

I don't think there is a straightforward solution to this or something VoxelShop should take ownership of. To me that seems like a custom workflow edge case.
